include/ur_api.h

```diff
@@ -5234,7 +5236,7 @@ typedef struct ur_kernel_native_properties_t {
 ///     - ::UR_RESULT_ERROR_ADAPTER_SPECIFIC
 ///     - ::UR_RESULT_ERROR_INVALID_NULL_HANDLE
 ///         + `NULL == hContext`
-///         + `NULL == hProgram`
+///         + If `hProgram == NULL` and the implementation requires a valid program.
```

Comment:
Do we have a test for this with a backend that does return `INVALID_NULL_HANDLE`?

Reply:
I think the current backends could query the program associated with the kernel via getInfo. So, I don't think the current backends have a hard requirement for the program being passed. However, I am not sure of `native_cpu` position on this.

Reply:
it is a hard requirement for level zero as far as I can tell, I've added a check and an adapter specific test